tests/test_consdbClient.py

```diff
@@ -0,0 +1,140 @@
+        ],
+    )
+    responses.post(
+        "http://example.com/consdb/flex/bad_instrument/exposure/addkey",
+        status=404,
+        json={"message": "Unknown instrument", "value": "bad_instrument", "valid": ["latiss"]},
+    )
+    responses.post(
+        "http://example.com/consdb/flex/latiss/bas_obs_type/addkey",
+        status=404,
+        json={"message": "Unknown observation type", "value": "bad_obs_type", "valid": ["exposure"]},
+    )
+
+    assert client.add_flexible_metadata_key(instrument, obs_type, "foo", "bool", "bool key")["key"] == "foo"
+    assert client.add_flexible_metadata_key(instrument, obs_type, "bar", "int", "int key")["key"] == "int"
+    client.add_flexible_metadata_key("bad_instrument", obs_type, "error", "int", "instrument error")
+    client.add_flexible_metadata_key(instrument, "bad_obs_type", "error", "int", "obs_type error")
+
+
+@responses.activate
+def test_get_flexible_metadata_keys(client):
+    description = [["foo", "bool", "a"], ["bar", "int", "b"]]
+    responses.get(
+        "http://example.com/consdb/flex/latiss/exposure/schema",
+        json=description,
+    )
+    instrument = "latiss"
+    obs_type = "exposure"
+    assert client.get_flexible_metadata_keys(instrument, obs_type) == description
+
+
+@responses.activate
+def test_get_flexible_metadata(client):
+    results = {"bool_key": True, "int_key": 42, "float_key": 3.14159, "str_key": "foo"}
+    responses.get(
+        "http://example.com/consdb/flex/latiss/exposure/obs/271828",
+        json=results,
+    )
+    responses.get(
+        "http://example.com/consdb/flex/latiss/exposure/obs/271828?k=float_key", json={"float_key": 3.14159}
+    )
+    responses.get(
+        "http://example.com/consdb/flex/latiss/exposure/obs/271828?k=int_key&k=float_key",
+        json={"float_key": 3.14159, "int_key": 42},
+    )
+    instrument = "latiss"
+    obs_type = "exposure"
+    obs_id = 271828
+    assert client.get_flexible_metadata(instrument, obs_type, obs_id) == results
+    assert client.get_flexible_metadata(instrument, obs_type, obs_id, ["float_key"]) == {
+        "float_key": results["float_key"]
+    }
+    assert client.get_flexible_metadata(instrument, obs_type, obs_id, ["int_key", "float_key"]) == {
+        "int_key": results["int_key"],
+        "float_key": results["float_key"],
+    }
+
+
+@responses.activate
+def test_schema(client):
+    description = [["foo", "bool", "a"], ["bar", "int", "b"]]
+    responses.get(
+        "http://example.com/consdb/schema/latiss/misc_table",
+        json=description,
+    )
+    instrument = "latiss"
+    table = "misc_table"
+    assert client.schema(instrument, table) == description
+
+
+# TODO: more POST tests
+#    client.insert_flexible_metadata(
+#        instrument, obs_type, obs_id, values, allow_update
+#    )
+#    client.insert(instrument, table, obs_id, values, allow_update)
+#    client.query(query)
```

Comment:
Uncomment or remove?

Reply:
TODOs (future tests to be added, in this case) are allowed, I think.